internal/exec/describe_affected.go

```diff
@@ -0,0 +1,326 @@
+package exec
+
+import (
+	"fmt"
+	cfg "github.com/cloudposse/atmos/pkg/config"
+	u "github.com/cloudposse/atmos/pkg/utils"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/tcnksm/go-gitconfig"
+	"os"
+	"path"
+	"reflect"
+	"strconv"
+	"time"
+)
+
+// ExecuteDescribeAffectedCmd executes `describe affected` command
+func ExecuteDescribeAffectedCmd(cmd *cobra.Command, args []string) error {
+	info, err := processCommandLineArgs("", cmd, args)
+	if err != nil {
+		return err
+	}
+
+	cliConfig, err := cfg.InitCliConfig(info, true)
+	if err != nil {
+		u.PrintErrorToStdError(err)
+		return err
+	}
+
+	// Process flags
+	flags := cmd.Flags()
+
+	ref, err := flags.GetString("ref")
+	if err != nil {
+		return err
+	}
+
+	sha, err := flags.GetString("sha")
+	if err != nil {
+		return err
+	}
+
+	if ref == "" && sha == "" {
```

Comment:
Should we consider setting the default to the head of the default branch if neither is specified rather than throwing an error?

Reply:
> Both can be specified

This is also confusing to me...if I specify a SHA shouldn't I get all changes on my current branch since that SHA? What happens if I specify both REF and SHA?

Reply:
This is still somewhat confusing to me...with regular git, if you checkout a commit, the branch is not necessary. It checks out the repository's state at that specific sha. The sha IS a ref just like the HEAD is an alias for whatever sha the main branch is currently pointing to. It seems to me that `ref` and `sha` shouldn't ever both be specified.